public/apps/configuration/panels/audit-logging/audit-logging.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import {
+  EuiDescribedFormGroup,
+  EuiForm,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiPanel,
+  EuiSwitch,
+  EuiTitle,
+} from '@elastic/eui';
+import { AppDependencies } from '../../../types';
+
+function renderStatusPanel(onSwitchChange, isSwitchChecked: boolean) {
+  return (
+    <EuiForm>
+      <EuiDescribedFormGroup
+        title={<h3>Enable audit logging</h3>}
+        description={<>Enable or disable audit logging</>}
+      >
+        <EuiFormRow>
+          <EuiSwitch
+            name="switch"
+            label={isSwitchChecked ? 'Enabled' : 'Disabled'}
+            checked={isSwitchChecked}
+            onChange={onSwitchChange}
+          />
+        </EuiFormRow>
+      </EuiDescribedFormGroup>
+    </EuiForm>
+  );
+}
+
+export function AuditLogging(props: AppDependencies) {
+  const [isSwitchChecked, setIsSwitchChecked] = useState(false);
+
+  const onSwitchChange = () => {
+    setIsSwitchChecked(!isSwitchChecked);
```

Comment:
The value of `isSwitchChecked` could be stale and not as expected. Always use functional state updates in such scenario. More details [here](https://reactjs.org/docs/hooks-reference.html#functional-updates)
```
setIsSwitchChecked(prevChcked => !prevChecked)
```